tool/resources/org/antlr/v4/tool/templates/codegen/Java/Java.stg

```diff
@@ -380,7 +416,7 @@ case <index>:
  * overriding implementation impossible to maintain.
  */
 RuleSempredFunction(r, actions) ::= <<
-private boolean <r.name>_sempred(<r.ctxType> _localctx, int predIndex) {
+private boolean <r.name>_sempred(<if(parser.file.genLean)><parser.name>Contexts.<endif><r.ctxType> _localctx, int predIndex) {
```

Comment:
@parrt `RuleSempredFunction` is used in both `Lexer` and `Parser` contexts. What would be a clean way to avoid a ST warning here ?

Reply:
Not sure what the warning is, but I'm sure you thought carefully about it